bitcoin/src/p2p/mod.rs

```diff
@@ -123,6 +122,7 @@ impl ServiceFlags {
 impl fmt::LowerHex for ServiceFlags {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { fmt::LowerHex::fmt(&self.0, f) }
 }
+impl_to_hex_from_lower_hex!(ServiceFlags, |_| 8);
```

Comment:
I've just noticed this doesn't zero-prefix the value. If this is desired behavior we should use `8 - flags.0.leading_zeros() / 8`.

Reply:
if zero-prefixing is desired, i think it should be done in the `core::fmt::LowerHex` implementation so that the macro doesn't have to handle that logic

Reply:
I don't know and that'd be for a separate PR but you should still pass the correct value to the macro (which I wrote above).

Reply:
Will do. Been swamped lately but will try to put something out in the next couple of days